### PR TITLE
Add support to propagate labels from NodePools to Nodes

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
This introduces inital support for label propagation from NodePools to Nodes honouring the following functional reqs:
- Propagate only once on Node creation. In future we might enable further automated or controller reconciliation.
- Enable consumers to choose the labels to be synced.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Ref https://issues.redhat.com/browse/HOSTEDCP-262

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by csrwng — COMMENTED** [review]
A couple of comments, otherwise lgtm

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -1280,6 +1282,13 @@ func (r *NodePoolReconciler) reconcileMachineDeployment(log logr.Logger,
 		},
 	}
 
+	for k, v := range nodePool.Spec.NodeLabels {
+		// Propagated managed labels down to Machines with a known hardcoded prefix
+		// so the CPO HCCO Node controller can recongnise them and apply them to Nodes.
+		labelKey := fmt.Sprintf("%s/%s", labelManagedPrefix, k)
```
[the remark this answers is not in the document]
Comment (enxebre) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:1288]: This prefix it's just to programatically recognise what needs to be synced. We trimmed it back before setting it as a label https://github.com/openshift/hypershift/pull/2026/files#diff-78c5e223954bb1f7b507c2e6f7622535548196670b9deb813f08a7a2335e4c95R93

[the remark this answers is not in the document]
Comment (enxebre) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:1288]: Oh but you're right it'd fail when set in the Machine itself

**Comment by csrwng** [comment]
@enxebre looks like there's a compilation error in the e2e test (`make e2e`)

**Comment by enxebre** [comment]
/hold
to add taints

File: api/v1alpha1/nodepool_types.go [hunk 2]
```diff
@@ -773,3 +781,17 @@ type NodePoolCondition struct {
 	// +kubebuilder:validation:Minimum=0
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 }
+
+// Taint is as v1 Core but without TimeAdded.
+// https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L3037-L3053
```
Comment (csrwng) [inline api/v1alpha1/nodepool_types.go:786]: nit: use a permanent link here

**Review by csrwng — COMMENTED** [review]
Two very small nits

**Pull request merged**